Kernel/Devices/Device.h

```diff
@@ -65,6 +65,8 @@ class Device : public File {
     static void for_each(Function<void(Device&)>);
     static Device* get_device(unsigned major, unsigned minor);
 
+    String device_name() const;
```

Comment:
Let's make this virtual and have each `Device` subclass determine its own name format instead of centralizing it in the base class.

Reply:
Done, but I'm not sure if I got the PTY stuff right... There's probably some more refactoring to be done here, but I'd rather not stack yet more things on top of this PR if I can help it.